src/OrchardCore/OrchardCore.Abstractions/Routing/RouteEndpointKey.cs

```diff
@@ -0,0 +1,62 @@
+using Microsoft.AspNetCore.Routing;
+
+namespace OrchardCore.Routing;
+
+public sealed class RouteEndpointKey : IEquatable<RouteEndpointKey>
```

Comment:
This could be a struct to improve performance slightly.

Reply:
I will agree only because we make these keys on each method call. Changed.